sys/kern/main.c

```diff
@@ -57,24 +70,19 @@ static __noreturn void start_init(__unused void *arg) {
   error = session_enter(p);
   assert(error == 0);
 
-  /* ... and initialize file descriptors required by the standard library. */
-  int _stdin, _stdout, _stderr;
-  do_open(p, "/dev/uart", O_RDONLY, 0, &_stdin);
-  do_open(p, "/dev/uart", O_WRONLY, 0, &_stdout);
-  do_open(p, "/dev/uart", O_WRONLY, 0, &_stderr);
-
-  assert(_stdin == 0);
-  assert(_stdout == 1);
-  assert(_stderr == 2);
-
   char *test = kenv_get("test");
-  if (test)
+  if (test) {
+    open_fds();
     ktest_main(test);
+  }
 
   char *init = kenv_get("init");
   if (init == NULL)
     init = "/sbin/init";
 
+  if (strcmp("/sbin/init", init))
```

Comment:
Please describe here that it is a workaround and what it is for.

Reply:
Added a comment in a19a4a7f2d2056822227eb9c628abe015471b88c